src/OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.cpp

```diff
@@ -0,0 +1,157 @@
+
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/CoordinatesBroker.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace trajectory
+{
+
+CoordinatesBroker::CoordinatesBroker()
+    : nextCoordinatesSubsetIndex_(0),
+      coordinatesSubsets_({}),
+      coordinatesSubsetsIndexMap_({})
+{
+}
+
+CoordinatesBroker::CoordinatesBroker(const Array<Shared<const CoordinatesSubset>>& aCoordinatesSubsetsArray)
+    : CoordinatesBroker()
+{
+    for (Shared<const CoordinatesSubset> subset : aCoordinatesSubsetsArray)
+    {
+        this->addSubset(subset);
+    }
+}
+
+bool CoordinatesBroker::operator==(const CoordinatesBroker& aCoordinatesBroker) const
+{
+    if (this->getNumberOfCoordinates() != aCoordinatesBroker.getNumberOfCoordinates())
+    {
+        return false;
+    }
+
+    if (this->getNumberOfSubsets() != aCoordinatesBroker.getNumberOfSubsets())
+    {
+        return false;
+    }
+
+    for (const auto& element : this->coordinatesSubsetsIndexMap_)
+    {
+        const String id = element.first;
+        if (!aCoordinatesBroker.hasSubset(id))
+        {
+            return false;
+        }
+
+        if (element.second != aCoordinatesBroker.getSubsetIndex(id))
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+bool CoordinatesBroker::operator!=(const CoordinatesBroker& aCoordinatesBroker) const
+{
+    return !((*this) == aCoordinatesBroker);
+}
+
+Size CoordinatesBroker::getNumberOfCoordinates() const
+{
+    return this->nextCoordinatesSubsetIndex_;
+}
+
+Size CoordinatesBroker::getNumberOfSubsets() const
+{
+    return (size_t)this->coordinatesSubsetsIndexMap_.size();
+}
+
+Array<Shared<const CoordinatesSubset>> CoordinatesBroker::getSubsets() const
+{
+    return this->coordinatesSubsets_;
+}
+
+Index CoordinatesBroker::addSubset(const Shared<const CoordinatesSubset>& aCoordinatesSubsetSPtr)
+{
+    const auto search = this->coordinatesSubsetsIndexMap_.find(aCoordinatesSubsetSPtr->getId());
+
+    if (search != this->coordinatesSubsetsIndexMap_.end())
+    {
+        return search->second;
+    }
+
+    Index preAdditionNextCoordinatesSubsetIndex = this->nextCoordinatesSubsetIndex_;
+
+    this->coordinatesSubsets_.add(aCoordinatesSubsetSPtr);
+    this->coordinatesSubsetsIndexMap_.insert({aCoordinatesSubsetSPtr->getId(), this->nextCoordinatesSubsetIndex_});
+    this->nextCoordinatesSubsetIndex_ += aCoordinatesSubsetSPtr->getSize();
+
+    return preAdditionNextCoordinatesSubsetIndex;
+}
+
+bool CoordinatesBroker::hasSubset(const Shared<const CoordinatesSubset>& aCoordinatesSubsetSPtr) const
+{
+    return this->hasSubset(aCoordinatesSubsetSPtr->getId());
+}
+
+Index CoordinatesBroker::getSubsetIndex(const Shared<const CoordinatesSubset>& aCoordinatesSubsetSPtr) const
+{
+    return this->getSubsetIndex(aCoordinatesSubsetSPtr->getId());
+}
+
+VectorXd CoordinatesBroker::extract(const VectorXd& allCoordinates, const CoordinatesSubset& aCoordinatesSubset) const
```

Comment:
Should we be providing the `CoordinatesSubset` in input or its idea? 🤔

Reply:
Checking further in the PRs to get a better idea

Reply:
Sorry didn't ge the `or its idea` part